### DISCUSSION TITLE
Using onboard flash of Riverdi display

### DISCUSSION THREAD

**Discussion by MrRedKite** [RudolphRiedel/FT800-FT813#76] [Q&A] [answered]
Hello,

im using the RVT70HSBNWC00 from Riverdi and a same51n20a.
The display works already and I want so save images in the onboard flash memory of the display.
Do I have to use the connector (P3) on the display with the "EVE Asset Builder" or can I just write the data from my mcu via "EVE_cmd_flashprogram"?

Thanks

**Comment by RudolphRiedel** [comment, marked as answer]
Regarding P3:
"Engineers at Riverdi decided to place an additional, dedicated port on the EVE4 board for fast programming (writing the content) flash memory chip from an external source."
"This process is planned to be used during production so the end customer will be able to order pre-programmed EVE4 boards with graphic content of their own choice. The process requires a special cable with a connector compatible with the fast programming port. The below photo shows the EVE4 board with fast-programming port (P3)."

So that footprint on the board probably is used with a Tag-Connect TC2050 connector and likely has pins for direct access to the QSPI of the Flash while driving the PD pin of the BT817 to low.  
No idea what software/hardware beyond the cable would be necessary, what they call "a special cable" should include some sort of USB interface.
Interesting is "planned" and "production", so they might not ever release documentation for it and the necessary equipment is likely rather expensive.

EAB would be using P1 over a board like the Riverdi Hermes: https://riverdi.com/blog/hermes-board-spi-usb-bridge-board-2/
And looks like Matrix Orbital released a new version of their board: https://www.matrixorbital.com/interface-module/eve-usb2spi-kit-b - this one has a switching regulator - nice, I would like to buy one.

But yes, you can also EVE_cmd_flashprogram() or EVE_cmd_flashupdate() from your controller.
My example shows how to write a flash-image from the controller to the chip.
Writing single picture files is not really conveniant to do though.
First you always need the "binary blob" in the first 4k of the flash that EAB automatically puts in from of every flash-image.
And then you need to take care of the alignment requirements yourself while EAB just deals with it.
When using EVE_cmd_flashupdate() everything is stored in RAM_G first so the uncompressed image file must not exceed 1 MiB.

**Reply by MrRedKite** [reply]
Thanks for the fast and detailed answer!
That is everything I wanted to know and even more ;)
